app/src/main/java/com/sci/recipeandroid/feature/auth/ui/viewmodel/SignUpViewModel.kt

```diff
@@ -0,0 +1,160 @@
+
+    private val _signUpScreenEvent = SingleLiveEvent<SignUpScreenEvent>()
+    val signUpScreenEvent: LiveData<SignUpScreenEvent> = _signUpScreenEvent
+
+    fun onEvent(event: RegistrationFormEvent) {
+        when (event) {
+            is RegistrationFormEvent.NameChanged -> {
+                _formState.value = _formState.value?.copy(displayName = event.displayName)
+            }
+
+            is RegistrationFormEvent.EmailChanged -> {
+                _formState.value = _formState.value?.copy(email = event.email)
+            }
+
+            is RegistrationFormEvent.PasswordChanged -> {
+                _formState.value = _formState.value?.copy(password = event.password)
+            }
+
+            is RegistrationFormEvent.RepeatedPasswordChanged -> {
+                _formState.value = _formState.value?.copy(repeatedPassword = event.repeatedPassword)
+            }
+
+            is RegistrationFormEvent.AcceptTerms -> {
+                _formState.value = _formState.value?.copy(acceptedTerms = event.isAccepted)
+            }
+
+            is RegistrationFormEvent.Submit -> {
+                submitData()
+            }
+
+        }
+    }
+
+    fun clearEmailError() {
+        _formState.value = _formState.value?.copy(emailError = null)
+    }
+
+    fun clearPasswordError() {
+        _formState.value = _formState.value?.copy(passwordError = null)
+    }
+
+    fun clearRepeatedPasswordError() {
+        _formState.value = _formState.value?.copy(repeatedPasswordError = null)
+    }
+
+    private fun submitData() {
+        val currentState = _formState.value ?: return
+        val nameResult = if (currentState.displayName.isBlank()) {
+            ValidationResult(successful = false, errorMessage = "Your name can't be empty")
+        } else {
+            ValidationResult(successful = true)
+        }
+        val emailResult = validator.emailValidator(currentState.email)
+        val passwordResult = validator.passwordValidator(currentState.password)
+        val repeatedPasswordResult = validator.confirmPasswordValidator(
+            currentState.password, currentState.repeatedPassword
+        )
+
+        val hasError = listOf(
+            emailResult,
+            passwordResult,
+            repeatedPasswordResult,
+            nameResult
+        ).any { !it.successful }
+
+        if (hasError) {
+            _formState.value = currentState.copy(
+                displayNameError = nameResult.errorMessage,
+                emailError = emailResult.errorMessage,
+                passwordError = passwordResult.errorMessage,
+                repeatedPasswordError = repeatedPasswordResult.errorMessage,
+            )
+            return
+        } else {
+            postToServer(
+                name = currentState.displayName,
+                email = currentState.email,
+                password = currentState.password
+            )
+        }
+    }
+
+    private fun postToServer(
```

Comment:
signUp

Reply:
changed this function name 9baf80468f57d78d6c4b13f545e1ebaa10f9d59c